lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2685,6 +2696,58 @@ class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenConstantPadNdOp : public OpConversionPattern<AtenConstantPadNdOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenConstantPadNdOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+    Value self = adaptor.self();
+    auto type = self.getType().cast<RankedTensorType>();
+    int64_t rank = type.getRank();
+
+    Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+    if (!elementType.isa<mlir::FloatType>())
+      return op.emitError("unimplemented: non-floating point type");
+
+    // Pattern match against the op's original operands, because otherwise we
+    // will get the lowered version of the operands which is harder to pattern
+    // match.
+    SmallVector<int64_t> padInts;
+    if (!matchPattern(op.pad(), m_TorchConstantIntList(padInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int pad ranges");
+    uint64_t padRank = padInts.size() / 2;
+    if (padRank * 2 != padInts.size())
+      return rewriter.notifyMatchFailure(op, "pad range size is not even");
+    if (rank < 0 || padRank > (uint64_t)rank)
+      return rewriter.notifyMatchFailure(op, "padding exceeds tensor rank");
+
+    // low/high initialized with the dims that should not be padded
+    SmallVector<int64_t, 4> lowPadding(/*Size=*/rank - padRank, /*Value=*/0);
+    SmallVector<int64_t, 4> highPadding(/*Size=*/rank - padRank, /*Value=*/0);
+    // add requested padding - note op.pad() is higest dim first ordered pairs
+    // of low,high
+    for (uint64_t i = padRank; i > 0; --i) {
+      lowPadding.push_back(padInts[i * 2 - 2]);
+      highPadding.push_back(padInts[i * 2 - 1]);
+    }
+
+    Value castedValue =
+        rewriter.create<arith::TruncFOp>(loc, type.getElementType(), adaptor.value());
```

Comment:
the `convertScalarToDtype` helper can be used to covert the `value` type to tensor element type. 

Also, technically, the `elementType` need to be got from the converted result type, `getTypeConverter()->convertType(op.getType())`;.  This is not an issue for float because the element type of converted tensor type and the torch tensor type are the same. However, it would be different for integer tensor types.

Reply:
Thanks for the explanation!